src/Components/App/App.js

```diff
@@ -34,17 +60,35 @@ class App extends Component {
   }
 
   render() {
+
     return (
       <div className="App">
         <Route 
           exact path='/'>
-          <Homepage />
+          <Homepage 
+            getActivityData={this.getActivityData}
+            allMovies={this.state.movies}
+          />
+        </Route>
+
+        <Route
+          exact path='/:activity'
+          render={({ match }) => {
+            return <BrowsePage 
+            name={match.params.activity}
+            data={ this.state[match.params.activity]}
+          />
+          }}>
```

Comment:
Very clever use the match params here, well done

Reply:
It really helped me a lot when we struggled passing the `event` up from the `form`  component to the `app` component. I think using the event in `JS` in general is such a powerful tool.